### PR TITLE
Update documentation on integration testing

### PR DESCRIPTION
**Pull request by mjwolf**
## Proposed commit message
Update the documentation pages to improve the pipeline, system and policy testing pages. This adds additional information on writing and running these tests, and removes some outdated information.

Major changes are:
* Rewrite pipeline testing docs to add more information on how to write and run tests
* Rewrite system testing docs to add more information on how to write and run tests
* Add page for policy tests
* Add more info to the tips page
* Add a section on how to upload integrations to a cloud/on-prem instance with elastic-package
* Move TSDS guidelines page to be under the guideline index.

### CODE REVIEW

File: docs/extend/pipeline-testing.md [hunk 1]
```diff
@@ -3,193 +3,394 @@ mapped_pages:
...
+For more information on pipeline tests, refer to [https://github.com/elastic/elastic-package/blob/main/docs/howto/pipeline_testing.md](https://github.com/elastic/elastic-package/blob/main/docs/howto/pipeline_testing.md).
 
-## Conceptual process [pipeline-concepts]
+## Quick Start [pipeline-quickstart]
 
-Conceptually, running a pipeline test involves the following steps:
+```bash
+# Start Elasticsearch
+elastic-package stack up -d --services=elasticsearch
+$(elastic-package stack shellinit)
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md]: This step shouldn't be needed anymore.
```suggestion
```

File: docs/extend/pipeline-testing.md [hunk 2]
```diff
@@ -3,193 +3,394 @@ mapped_pages:
...
 
-1. Deploy the {{es}} instance (part of the {{stack}}). This step takes time, so you should typically do it once as a prerequisite to running pipeline tests on multiple data streams.
-2. Upload ingest pipelines to be tested.
-3. Use the [Simulate API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-ingest-simulate) to process logs/metrics with the ingest pipeline.
-4. Compare generated results with expected ones.
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md:18]: I am fine with removing these details, but I would still mention the use of the simulate API for these tests. Though this is also explained in the linked docs in elastic-package (https://github.com/elastic/elastic-package/blob/main/docs/howto/pipeline_testing.md).

File: docs/extend/pipeline-testing.md [hunk 3]
```diff
@@ -3,193 +3,394 @@ mapped_pages:
...
   - network.iana_number
 ```
 
-The `multiline` section [raw files only](#pipeline-raw-files) configures the log file reader to detect multiline log entries using the `first_line_pattern`. Use this property if you may split your logs into multiple lines, e.g. Java stack traces.
+### Multiline Configuration [multiline-config]
 
-The `fields` section allows for customizing extra fields to be added to every read log entry (e.g. `@timestamp`, `ecs`). Use this property to extend your logs with data that can’t be extracted from log content, but it’s fine to have the same field values for every record (e.g. timezone, hostname).
+For logs that span multiple lines:
 
-The `dynamic_fields` section allows for marking fields as dynamic (every time they have different non-static values), so that pattern matching instead of strict value check is applied.
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md]: `fields` and `dynamic_fields` are not explained now?

File: docs/extend/integrations-guidelines.md [hunk 4]
```diff
@@ -10,7 +10,4 @@ Refer to the following pages for some tips and recommendations for building inte
 * [General guidelines](/extend/general-guidelines.md)
 * [Dashboard guidelines](/extend/dashboard-guidelines.md)
 * [Documentation guidelines](/extend/documentation-guidelines.md)
-
-
-
-
+* [TSDS guidelines](/extend/developer-tsds-guidelines.md))
```
Comment (mrodm) [inline docs/extend/integrations-guidelines.md]: Extra ")" ?

```suggestion
* [TSDS guidelines](/extend/developer-tsds-guidelines.md)
```

File: docs/extend/policy-testing.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+For more details, see [HOWTO: Writing policy tests for a package](https://github.com/elastic/elastic-package/blob/main/docs/howto/policy_testing.md)
+
+## Running policy tests [policy-tests]
+
+Policy tests don’t require the {{stack}} to be up and running. Simply navigate to the package’s root folder (or any sub-folder under it) and run the following command.
+
+```bash
+elastic-package test policy
+```
```
Comment (mrodm) [inline docs/extend/policy-testing.md]: IIRC policy tests require to be the stack up and running.

Just tested locally, and without the stack running, `elastic-package` cannot run these tests:

```bash
 $ elastic-package stack status
Status of Elastic stack services:
 - No service running
 $ elastic-package test policy 
Run policy tests for the package
Error: can't create Kibana client: the Elastic stack is unavailable, remember to start it with 'elastic-package stack up', or configure elastic-package with environment variables
```

File: docs/extend/integrations-guidelines.md (hunk 4 above)
Comment (mjwolf) [reply docs/extend/integrations-guidelines.md]: yes, removed

File: docs/extend/pipeline-testing.md (hunk 1 above)
Comment (mjwolf) [reply docs/extend/pipeline-testing.md]: Thanks, I’ve removed this from the file and all the other testing files that used it.

File: docs/extend/pipeline-testing.md (hunk 2 above)
Comment (mjwolf) [reply docs/extend/pipeline-testing.md:18]: I've added this back into a new "Under the hood" section at the end of the file.

I think this is detail that isn't really needed when starting to use pipeline tests, and might be confusing to new users, so I put it at the bottom of the file as more of an advanced info section

File: docs/extend/pipeline-testing.md (hunk 3 above)
Comment (mjwolf) [reply docs/extend/pipeline-testing.md]: I've added back explainations of these

File: docs/extend/policy-testing.md (hunk 5 above)
Comment (mjwolf) [reply docs/extend/policy-testing.md]: I've changed this to state you need the stack running now.

File: docs/extend/pipeline-testing.md [hunk 6]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
-1. Deploy the {{es}} instance (part of the {{stack}}). This step takes time, so you should typically do it once as a prerequisite to running pipeline tests on multiple data streams.
-2. Upload ingest pipelines to be tested.
-3. Use the [Simulate API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-ingest-simulate) to process logs/metrics with the ingest pipeline.
-4. Compare generated results with expected ones.
+# Run pipeline tests
+cd packages/your-package
+elastic-package test pipeline
 
+# Generate expected results (first time setup)
+elastic-package test pipeline --generate
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md:24]: Should we be more explicit on when the `--generate` flag should be used?

File: docs/extend/pipeline-testing.md [hunk 7]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
-* As you’re only testing the ingest pipeline, you can prepare mocked documents with imaginary fields, different from ones collected in {{beats}}. Also, the other way round, you can skip most of the example fields and use tiny documents with a minimal set of fields just to satisfy the pipeline validation.
-* There might be integrations that transform data mainly using {{beats}} processors instead of ingest pipelines. In such cases, ingest pipelines are rather plain.
+## What Pipeline Tests Validate [pipeline-validation]
 
+Pipeline tests verify:
+- Field extraction and parsing logic
+- Data type conversions and formatting
+- ECS field mapping compliance
+- Error handling and edge cases
+- Performance of complex processors
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md]: Well, for performance there is another subcommand, `elastic-package benchmark pipeline`. These benchmarks can be executed once there are pipeline tests, so we could maybe mention this here. Though current implementation is not very accurate unless the tests are feed with lots of documents.

File: docs/extend/pipeline-testing.md [hunk 8]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
+            test-sample.log-expected.json  # Expected output
+            test-events.json               # JSON event input
+            test-events.json-expected.json # Expected output
 ```
 
+## Test Input Types [input-types]
 
-### Test case definitions [pipeline-test-case]
-
-There are two types of test case definitions - **raw files** and **input events**.
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md:57]: Should we mention that they are differentiated by the extension? Raw log files must have the `.log` extension, JSON event files must have the `.json` extension.

File: docs/extend/pipeline-testing.md [hunk 9]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
 
+# 5. Iterate on pipeline, then regenerate when needed
+elastic-package test pipeline --data-streams your-stream --generate
+```
+
+### Troubleshooting [troubleshooting]
+
+**Common issues and solutions:**
+
+**Test failures with field mismatches:**
```
Comment (jsoriano) [inline docs/extend/pipeline-testing.md]:
```suggestion
**Test failures with field value mismatches:**
```

File: docs/extend/policy-testing.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+---
+mapped_pages:
+  - https://www.elastic.co/guide/en/integrations-developer/current/policy-testing.html
+---
+
+# Policy testing [policy-testing]
+
+Policy tests allow you to verify that an integration policy will be accepted by Fleet, and will create an expect Elastic Agent policy.
```
Comment (jsoriano) [inline docs/extend/policy-testing.md]:
```suggestion
Policy tests allow you to verify that an integration policy will be accepted by Fleet, and will create an expected Elastic Agent policy.
```

File: docs/extend/policy-testing.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+For more details, see [HOWTO: Writing policy tests for a package](https://github.com/elastic/elastic-package/blob/main/docs/howto/policy_testing.md)
+
+## Running policy tests [policy-tests]
+
+Policy tests require the {{stack}} to be running. Start the Elastic stack with elastic-package, if it's not already running. Then simply navigate to the package’s root folder (or any sub-folder under it) and run the following command.
+
+```bash
+# Start the stack (if not already running)
+elastic-package stack up -d
```
Comment (jsoriano) [inline docs/extend/policy-testing.md]: Only kibana and elasticsearch are needed for these tests.
```suggestion
elastic-package stack up -d --services kibana,elasticsearch
```

File: docs/extend/system-testing.md [hunk 12]
```diff
@@ -3,242 +3,395 @@ mapped_pages:
...
+- Pipeline processing and field mappings
+- Document indexing into Elasticsearch data streams
+- Field type compatibility and mapping validation
+- Integration configuration and policy deployment
 
+The test framework automatically:
+1. Deploys your integration service (Docker/Kubernetes/Terraform)
+2. Configures the Elastic Agent with test policies
+3. Collects and indexes sample data
+4. Validates the first 500 documents by timestamp
```
Comment (jsoriano) [inline docs/extend/system-testing.md]: This is actually configurable. By default it just wait to have _any_ document in the index, it can be configured to wait for more documents with `assert.hit_count`, `assert.min_count` or `assert.fields_present`.

File: docs/extend/system-testing.md [hunk 13]
```diff
@@ -3,242 +3,395 @@ mapped_pages:
...
-| Placeholder name | Data type | Description |
-| --- | --- | --- |
-| `Hostname` | string | Addressable host name of the integration service. |
-| `Ports` | []int | Array of addressable ports the integration service is listening on. |
-| `Port` | int | Alias for `Ports[0]`. Provided as a convenience. |
-| `Logs.Folder.Agent` | string | Path to integration service’s logs folder, as addressable by the Agent. |
-| `SERVICE_LOGS_DIR` | string | Alias for `Logs.Folder.Agent`. Provided as a convenience. |
+# Verify stack is running
+elastic-package stack status
+```
```
Comment (jsoriano) [inline docs/extend/system-testing.md:286]: Maybe at this point we could link to https://github.com/elastic/elastic-package/blob/main/docs/howto/system_testing.md#test-case-definition for the full list of options that can be used on test definitions.

File: docs/extend/tips-for-building.md [hunk 14]
```diff
@@ -5,53 +5,513 @@ mapped_pages:
...
+    format: rfc5424
+{{#if tz_offset}}
+    timezone: "{{tz_offset}}"
+{{/if}}
+{{#if processors}}
+{{processors}}
+{{/if}}
+```
+
+### More Detail
```
Comment (jsoriano) [inline docs/extend/tips-for-building.md]: Nit. 
```suggestion
### More detail
```

File: docs/extend/tips-for-building.md [hunk 15]
```diff
@@ -5,53 +5,513 @@ mapped_pages:
...
+
+To export a dashboard for inclusion in the integration, use `elastic-package export dashboards`. Name the dashboard with a consistent pattern like `[Integration Name] Overview`. Adding `-- export this one` to the name can make it easier to find when exporting.
+
+When running `elastic-package export dashboards`, it will list all dashboards and allow filtering. Use arrow keys to navigate, spacebar to select, and enter to confirm. The exported dashboard will be saved as a file like `kibana/dashboards/integration-88888888-4444-4444-4444-cccccccccccc.json`, with an actual UUID. Edit the file to remove any `-- export this one` from the title.
+
+If edits to the dashboard are needed later, the file may need to be completely replaced with a new export, as all the UUIDs will change if cloning is required.
+
+When in the dashboard view, create a data filter, otherwise `elastic-package check` will fail the dashboard. Next to the `KQL` search box is a `+` button for this purpose. A good initial filter is `data_stream.dataset : integration-name.log`.
+
```
Comment (jsoriano) [inline docs/extend/tips-for-building.md:376]: Maybe we should mention `elastic-package edit dashboards`, that can be used to edit existing dashboards. Dashboards are installed read-only, and this subcommand removes this flag so they can be edited and exported.

File: docs/extend/pipeline-testing.md [hunk 16]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
 
 ```bash
+# Start only Elasticsearch (faster than full stack)
+elastic-package stack up -d --services=elasticsearch
+
+# Set environment variables
+$(elastic-package stack shellinit)
+
+# Verify Elasticsearch is running
+curl -X GET "localhost:9200/_cluster/health"
```
Comment (mrodm) [inline docs/extend/pipeline-testing.md]: `elastic-package` starts the services with HTTPS enabled, so these commands should be using `https://` too.

Then it could be used --insecure in curl commands or the part to the cert 
- using --insecure / -k
```
curl -u elastic:changeme -k \
  -X GET \
  "https://localhost:9200/_cluster/health"
```
- using --cacert (assuming `default` profile)
```
curl -u elastic:changeme \
  --cacert ~/.elastic-package/profiles/default/stack/certs/ca-cert.pem \
  -X GET \
  "https://localhost:9200/_cluster/health"
```

File: docs/extend/pipeline-testing.md [hunk 17]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
+
+# Check pipeline syntax
+elastic-package lint
 
-Next, you must set environment variables needed for further `elastic-package` commands.
+# Manually test pipeline upload
+curl -X PUT "localhost:9200/_ingest/pipeline/your-pipeline" \
+  -H "Content-Type: application/json" \
+  -d @data_stream/your-stream/elasticsearch/ingest_pipeline/default.yml
+```
```
Comment (mrodm) [inline docs/extend/pipeline-testing.md]: Same as above. It should be added https, and the required curl parameters:
- Using --insecure / -k 
```
curl -u elastic:changeme -k -X PUT "localhost:9200/_ingest/pipeline/your-pipeline" \
  -H "Content-Type: application/json" \
  -d @data_stream/your-stream/elasticsearch/ingest_pipeline/default.yml
```
- Using --cacert
```
curl -u elastic:changeme --cacert ~/.elastic-package/profiles/default/certs/elasticsearch/ca-cert.pem -X PUT "localhost:9200/_ingest/pipeline/your-pipeline" \
  -H "Content-Type: application/json" \
  -d @data_stream/your-stream/elasticsearch/ingest_pipeline/default.yml
```

File: docs/extend/pipeline-testing.md [hunk 18]
```diff
@@ -3,193 +3,407 @@ mapped_pages:
...
 
 ```bash
-elastic-package test pipeline --data-streams <data stream 1>[,<data stream 2>,...]
+# Test individual pipeline components
+curl -X POST "localhost:9200/_ingest/pipeline/_simulate" \
+  -H "Content-Type: application/json" \
+  -d '{
+    "pipeline": {"processors": [{"grok": {"field": "message", "patterns": ["your-pattern"]}}]},
+    "docs": [{"_source": {"message": "test log line"}}]
+  }'
```
Comment (mrodm) [inline docs/extend/pipeline-testing.md]: Same as above comments (https and other curl parameters)

File: docs/extend/tips-for-building.md [hunk 19]
```diff
@@ -5,53 +5,513 @@ mapped_pages:
...
+elastic-package stack up -vd [--version VERSION]
+	Bring the stack up. "-vd" is short for verbose output and detach from containers when done. Specify version if desired, such as '--version 8.12.1'.
 
+elastic-package stack down
+	Bring down the stack. Destroys containers.
 
+elastic-package stack up -vd --services package-registry
+	Recreates the package-registry container. Use after the build command to make the registry aware of your new package. Beware: If you install the package in Kibana, you can no longer update the package at that version. Increment the package version to make new packages show up. Remember to revert the version back to the original before submitting a PR.
+
+elastic-package test [pipeline|test|static|asset] -v
```
Comment (mrodm) [inline docs/extend/tips-for-building.md]: Missing subcommands for test:
```suggestion
elastic-package test [pipeline|static|asset|policy|system] -v
```

File: docs/extend/tips-for-building.md [hunk 20]
```diff
@@ -5,53 +5,513 @@ mapped_pages:
...
+    Run package tests. Make sure you are in the package's directory. A stack needs to be running for this to work. It is not necessary to build the package for the tests. Pipeline tests are great for rapid iteration given how quickly they run and how comprehensive the validations are. System tests are great for end-to-end tests and validating any changes made to Filebeat (this includes the *.yml.hbs files in data_stream/NAME/agent/stream).
+
+elastic-package test [pipeline|test|static|asset] -v -g
```
Comment (mrodm) [inline docs/extend/tips-for-building.md]: Missing subcommands for test:
```suggestion
elastic-package test [pipeline|static|asset|policy|system] -v -g
```
